trex.org

```diff
@@ -732,36 +738,36 @@ prim_factor =
   the basis of Slater determinants.
 
   #+NAME: csf
-  | Variable          | Type            | Dimensions                  | Description                                    |
-  |-------------------+-----------------+-----------------------------+------------------------------------------------|
-  | ~num~             | ~dim readonly~  |                             | Number of CSFs                                 |
-  | ~coefficient~     | ~float special~ | ~(state.num,csf.num)~       | Coefficients of the CSFs from the CI expansion |
-  | ~det_coefficient~ | ~float sparse~  | ~(determinant.num,csf.num)~ | Projection on the determinant basis            |
+  | Variable          | Type           | Dimensions                  | Description                                    |
+  |-------------------+----------------+-----------------------------+------------------------------------------------|
+  | ~num~             | ~dim readonly~ |                             | Number of CSFs                                 |
+  | ~coefficient~     | ~float buffer~ | ~(csf.num)~                 | Coefficients of the CSFs from the CI expansion |
+  | ~det_coefficient~ | ~float sparse~ | ~(determinant.num,csf.num)~ | Projection on the determinant basis            |
 
   #+CALL: json(data=csf, title="csf")
 
   #+RESULTS:
   :results:
   #+begin_src python :tangle trex.json
       "csf": {
-                      "num" : [ "dim readonly" , []                               ]
-        ,     "coefficient" : [ "float special", [ "csf.num", "state.num" ]       ]
-        , "det_coefficient" : [ "float sparse" , [ "csf.num", "determinant.num" ] ]
+		      "num" : [ "dim readonly", []                               ]
+	,     "coefficient" : [ "float buffer", [ "csf.num" ]                    ]
+	, "det_coefficient" : [ "float sparse", [ "csf.num", "determinant.num" ] ]
       } ,
   #+end_src
   :end:
 
 * Excited states (state group)
 
-  By default, the ~determinant~ group corresponds to the ground state.
-  However, it should be also possible to store the coefficients that
-  correspond to excited state wave functions for the same set of
-  determinants. This is the goal of the present group
+  This group contains information about excited state. Since TREXIO version 2.3.0
+  the state-specific data (e.g. CI/CSF coeffcients, RDMs) is written in a separate
+  file in order to avoid over-complicated internal logics and global state switches.
 
   #+NAME: state
   | Variable | Type  | Dimensions    | Description                                    |
   |----------+-------+---------------+------------------------------------------------|
   | ~num~    | ~dim~ |               | Number of states (including the ground state)  |
+  | ~id~     | ~int~ |               | Index of a current state (0 is ground state)   |
   | ~label~  | ~str~ | ~(state.num)~ | Label of a given state (e.g. 'S' for singlets) |
```

Comment:
So now, label should not be an array any more, but just a single string to describe the state in the current file.

Reply:
> I would even consider introducing an additional str array here with state.num dimension which will contain names of files with excited state data.

It is a very good idea to add a list of additional files in the metadata section.

I would however leave the description of the state in the corresponding file.

Reply:
OK. However, to avoid breaking the API over this minor change, I propose to keep this array as is and then add an additional string attribute called `current_label` which would indicated the label of the state in current file.